modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/Loader.scala

```diff
@@ -151,21 +156,31 @@ object Loader {
     loading.handleErrorWith(reportLoadFailure[F](discovery, addFailure))
   }
 
-  private def addLoadTstampColumn[F[_]: DAO: Monad: Logging](targetConfig: StorageTarget): F[Unit] =
-    targetConfig match {
-      // Adding load_tstamp column explicitly is not needed due to merge schema
-      // feature of Databricks. It will create missing column itself.
-      case _: StorageTarget.Databricks => Monad[F].unit
-      case _ =>
-        for {
-          allColumns <- DbControl.getColumns[F](EventsTable.MainName)
-          _ <- if (loadTstampColumnExist(allColumns))
-            Logging[F].info("load_tstamp column already exists")
-          else
-            DAO[F].executeUpdate(Statement.AddLoadTstampColumn, DAO.Purpose.NonLoading).void *>
-              Logging[F].info("load_tstamp column is added successfully")
-        } yield ()
+  private def addLoadTstampColumn[F[_]: Transaction[*[_], C]: Monitoring: Logging: MonadThrow,
+                                  C[_]: DAO: Monad: Logging](shouldAdd: Boolean, targetConfig: StorageTarget): F[Unit] = {
+    if (!shouldAdd) Logging[F].info("Adding load_stamp is skipped")
+    else {
+      val f = targetConfig match {
+        // Adding load_tstamp column explicitly is not needed due to merge schema
+        // feature of Databricks. It will create missing column itself.
+        case _: StorageTarget.Databricks => Monad[C].unit
+        case _ =>
+          for {
+            allColumns <- DbControl.getColumns[C](EventsTable.MainName)
+            _ <- if (loadTstampColumnExist(allColumns))
+              Logging[C].info("load_tstamp column already exists")
+            else
+              DAO[C].executeUpdate(Statement.AddLoadTstampColumn, DAO.Purpose.NonLoading).void *>
+                Logging[C].info("load_tstamp column is added successfully")
+          } yield ()
+      }
+      Transaction[F, C].transact(f).recoverWith {
+        case e: Throwable =>
+          val err = s"Error while adding load_tstamp column: $e"
```

Comment:
I know it's currently private, but it would be nice to [reuse this `getErrorMessage`](https://github.com/snowplow/snowplow-rdb-loader/blob/4.2.1/modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/Loader.scala#L214-L218)